src/ngraph/op/gather_tree.hpp

```diff
@@ -0,0 +1,58 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include "ngraph/op/op.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace v1
+        {
+            /// \brief Generates the complete beams from the ids per each step and the parent beam
+            /// ids.
+            class NGRAPH_API GatherTree : public Op
```

Comment:
AFAIK we don't use `NGRAPH_API` for other ops. So why do we keep it here?

Reply:
And in other ops it's usually used before type_infor

Reply:
@GlebKazantaev You are right but it was changed after https://github.com/NervanaSystems/ngraph/pull/3767 introduction